sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/TimeAndFormat.java

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A class to hold an {@code OffsetDateTime} object and the formatter from which it was parsed for effective
+ * roundtripping.
+ * Without this, we always format date-times the same way, but if we parsed a value from a sas token in on an endpoint
+ * or in a connection string, we have to preserve the formatting, especially the precision, so the signature still
+ * matches.
+ */
+public final class TimeAndFormat {
```

Comment:
Given this class isn't exposed in any public APIs does it itself need to be exposed as a public API?